concent_api/core/payments/bankster.py

```diff
@@ -384,23 +381,68 @@ def sum_subtask_price(subtask_results_accepted_list: List[SubtaskResultsAccepted
 
 def get_provider_payment_info(
     list_of_forced_payments: List[ForcedPaymentEvent],
-    list_of_payments: List[BatchTransferEvent],
+    already_satisfied_claims: QuerySet,
     subtask_results_accepted_list: List[SubtaskResultsAccepted],
 ) -> Tuple[int, int]:
-    assert isinstance(list_of_payments, list)
+    assert isinstance(already_satisfied_claims, QuerySet)
     assert isinstance(list_of_forced_payments, list)
     assert isinstance(subtask_results_accepted_list, list)
 
-    force_payments_price = sum_payments(list_of_forced_payments)
-    payments_price = sum_payments(list_of_payments)
+    forced_satisfied_payments_price = sum_payments(list_of_forced_payments)
+    satisfied_payments_price = already_satisfied_claims.aggregate(
+        sum_of_already_satisfied_claims=Coalesce(Sum('amount'), 0)
+    )['sum_of_already_satisfied_claims']
     subtasks_price = sum_subtask_price(subtask_results_accepted_list)
 
-    amount_paid = payments_price + force_payments_price
+    amount_paid = satisfied_payments_price + forced_satisfied_payments_price
```

Comment:
`payments_price` cannot be replaced with `satisfied_payments_price`. They're completely different things.

`payments_price` is `sum(LT)` - i.e. the amount the requestor has paid the normal way. If you remove it, you're ignoring all the non-forced payments.
`satisfied_payments_price + forced_satisfied_payments_price` is `sum(LF)`

After this change the code is computing `V` as `sum(LAR) - sum(LF)` instead of `sum(LAR) - sum(LT) - sum(LF)` as it should.

Reply:
:+1: done, I thought that (from issue):

> It uses both lists when checking existing settlement payments.

means that it should use *only* those two lists.

Reply:
Then I'd specifically write that it uses two lists :) I wrote that it uses **both** - i.e. both lists mentioned in the text above it, as opposed to the old code that was using just one of them.

Anyway, it should be clear from context anyway. Removing `LT` makes no sense in the context of goal stated in the issue - preventing multiple settlements by taking into account payments that are not yet on the blockchain. How would ignoring `LT` help us accomplish that?